### DISCUSSION TITLE
Option to add cache-control headers to pages with automatic static optimization when using `next start`

### DISCUSSION THREAD

**Discussion by flybayer** [vercel/next.js#64628] [Ideas]
### Describe the feature you'd like to request

Currently pages that are automatically statically optimized (no `getServerSideProps` or `getStaticProps` or `getInitialProps`) do not have any cache-control header set when served via `next start`.

This causes a slow down for all users of apps using `next start` because these static pages cannot be served from a CDN. Additionally it causes additional server load.

I want these pages to be automatically cached by any CDN in front of it.

### Describe the solution you'd like

I want a way to enable cache-control headers for auto static pages via next.config.js.

It could be a single flag like 
```js
module.exports = {
  automaticStaticOptimizationCacheControl: true | false | 's-max-age=36000'
}
```

Or potentially using the headers array with a special source.

```js
const config = require('next/config')

module.exports = {
  async headers() {
    return [
      {
        source: config.AUTOMATIC_STATIC_OPTIMIZATION,
        headers: [
          {
            key: 'Cache-control',
            value: 's-max-age=36000',
          },
        ],
      }
    ]
  },
}
```

And ideally we could enable it with an environment variable like `NEXT_ENABLE_STATIC_OPTIMIZATION_CACHE_CONTROL`

### Describe alternatives you've considered

I don't see any other alternatives besides manually configuring the CDN to cache specific pages or by using a custom server. Because the docs say "You cannot set Cache-Control headers in next.config.js"

**Comment by vivek7405** [comment]
It would be great if static pages can be auto served via CDN!

**Comment by AdamZaczek** [comment]
Our users are stuck forever on old versions of the app.
How can this be an intended behavior?
This is insanity. Vercel team, please take a look into this.

**Comment by Axxxx0n** [comment]
adding to that, static files from public dir have `cache-control: public, max-age=0` which is also bypassing cache.

**Comment by balazsorban44** [comment]
Converted to a discussion as per our [Triaging policy](https://github.com/vercel/next.js/blob/canary/contributing/repository/triaging.md). This does not mean that the discussion got closed, feel free to discuss further!

**Comment by leerob** [comment]
It's now possible to customize `cache-control` as of v14.2.10. Please upgrade and give it a try!

https://github.com/vercel/next.js/pull/69802
